### PR TITLE
feat: harden PDF comparison API and validation

### PR DESCRIPTION
**Pull request by dichovsky**
## Summary
- add a detailed comparison API, typed library errors, library-owned public types, and page-number-based comparison planning
- refactor PDF comparison into focused internal modules with explicit diff writing, input/output path guards, bounded-memory page rendering, and upgraded renderer/comparator dependencies
- strengthen the validation and release surface with broader regression coverage, built-artifact/type checks, contributor docs, and a dev/prod TypeScript config split

## What changed
- introduced `comparePdfDetailed()` and detailed result/page result types while keeping `comparePdf()` as a boolean wrapper
- added typed errors for input, configuration, rendering, and comparison failures
- replaced dependency-owned public types with library-owned `PdfInput`, `PdfRenderOptions`, `PageExclusion`, `PageArea`, and `RgbColor`
- moved orchestration into internal normalization/planning/render/comparison modules
- made diff writing explicit via `writeDiffs`, added diff root and input root enforcement, and keyed exclusions/planning by `pageNumber`
- changed rendering to metadata-first, page-at-a-time comparison to keep memory bounded while preserving the sequential renderer constraint
- upgraded to `png-visual-compare` 6.0 and `pdf-to-png-converter` 4.0.0, aligning the repo to Node 24 and handling renderer output-folder namespacing under `actual/` / `expected/`
- split TypeScript config into workspace `tsconfig.json` and production `tsconfig.prod.json`, removing `tsconfig.consumer.json`
- added changelog/contributing docs and expanded tests for published artifacts, types, errors, diff behavior, path guards, and internal edge cases

## Validation
- `npm test`

### CODE REVIEW

File: src/internal/renderPdfPages.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    return renderedPages[0];
+}
+
+function toRenderablePdfInput(pdfFile: PdfInput): string | Buffer | ArrayBuffer {
+    if (typeof pdfFile === 'string' || Buffer.isBuffer(pdfFile)) {
+        return pdfFile;
+    }
+
+    return Uint8Array.from(new Uint8Array(pdfFile)).buffer;
```
[the remark this answers is not in the document]
Comment (dichovsky) [reply src/internal/renderPdfPages.ts:72]: Looked into this, but pdf-to-png-converter 4.0 still detaches ArrayBuffer inputs across our sequential render calls, so returning ArrayBuffer as-is breaks the array-buffer tests. I kept the defensive copy to preserve correctness.
